litestar_asyncpg/config.py

```diff
@@ -88,10 +88,10 @@ class PoolConfig:
     max_inactive_connection_lifetime: float | EmptyType = Empty
     """Number of seconds after which inactive connections in the pool will be closed. Pass 0 to disable this mechanism."""
 
-    setup: Coroutine[None, None, Any | None] | EmptyType = Empty
-    """A coroutine to prepare a connection right before it is returned from Pool.acquire(). An example use case would be to automatically set up notifications listeners for all connections of a pool."""
-    init: Coroutine[None, None, Any | None] | EmptyType = Empty
-    """A coroutine to prepare a connection right before it is returned from Pool.acquire(). An example use case would be to automatically set up notifications listeners for all connections of a pool."""
+    setup: Callable[[Connection | PoolConnectionProxy], Coroutine[Any, Any, None]] | EmptyType = Empty
+    """A callable to prepare a connection right before it is returned from Pool.acquire(). An example use case would be to automatically set up notifications listeners for all connections of a pool."""
+    init: Callable[[Connection | PoolConnectionProxy], Coroutine[Any, Any, None]] | EmptyType = Empty
```

Comment:
I think for `init` it's only `Connection` and not `Connection | PoolConnectionProxy` ([ref](https://github.com/bryanforbes/asyncpg-stubs/blob/febc5aa878d3d09d690b4aa4ea05601b96fadf00/asyncpg-stubs/pool.pyi#L32-L33)).